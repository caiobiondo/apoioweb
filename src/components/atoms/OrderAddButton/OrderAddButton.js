import React from 'react';
import { FloatingActionButton, Icon } from 'natura-ui';
import { IconWrapper } from './OrderAddButton.styles';

const OrderAddButton = () => (
  <FloatingActionButton iconWrapper={IconWrapper}>
    <Icon file="ico_add_box" />
  </FloatingActionButton>
);

export default OrderAddButton;
