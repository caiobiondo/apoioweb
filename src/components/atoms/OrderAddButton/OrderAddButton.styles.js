import styled from 'styled-components';
import { white } from 'styles/colors';

export const IconWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;

  svg {
    fill: ${white};
    width: 30px;
    height: 30px;
  }
`;
