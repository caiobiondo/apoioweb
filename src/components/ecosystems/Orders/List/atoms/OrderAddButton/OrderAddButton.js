import React from 'react';
import { FloatingActionButton, Icon } from 'natura-ui';
import { IconWrapper } from './OrderAddButton.styles';

const OrderAddButton = () => (
  <FloatingActionButton onClick={() => redirectToOrders()} iconWrapper={IconWrapper}>
    <Icon file="ico_add_box" />
  </FloatingActionButton>
);

const redirectToOrders = () => {
  window.open('https://pedidos.natura.net/captaweb/');
};

export default OrderAddButton;
