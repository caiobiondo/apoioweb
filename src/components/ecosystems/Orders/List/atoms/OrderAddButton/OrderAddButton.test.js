import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import OrderAddButton from './OrderAddButton';

describe('OrderAddButton', () => {
  it('should render a floating button with add order icon', () => {
    const renderer = new ShallowRenderer();

    renderer.render(<OrderAddButton />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
