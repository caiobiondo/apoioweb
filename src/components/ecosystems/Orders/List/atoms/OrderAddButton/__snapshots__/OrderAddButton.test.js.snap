// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderAddButton should render a floating button with add order icon 1`] = `
<CustomFloatingActionButton
  iconWrapper={[Function]}
>
  <Icon
    file="ico_add_box"
  />
</CustomFloatingActionButton>
`;
